plugins/components/collider/Collider.hpp

```diff
@@ -64,20 +74,26 @@ namespace Components {
          * @throws std::runtime_error If the data size is invalid.
          */
         void deserialize(std::vector<uint8_t> &data) override {
-            if (data.size() != getConstantSize())
+            if (data.size() != getSize())
                 throw std::runtime_error("Invalid data size for Collider component");
 
-            width = *reinterpret_cast<float *>(&data[0]);
-            height = *reinterpret_cast<float *>(&data[sizeof(float)]);
-        };
+            uint32_t widthBits = (data[0] << 24) | (data[1] << 16) | (data[2] << 8) | data[3];
+            uint32_t heightBits = (data[4] << 24) | (data[5] << 16) | (data[6] << 8) | data[7];
+
+            widthBits = ntohl(widthBits);
+            heightBits = ntohl(heightBits);
+
+            width = *reinterpret_cast<uint32_t *>(&widthBits);
+            height = *reinterpret_cast<uint32_t *>(&heightBits);
```

Comment:
I have no idea why you're assigning from the address here though, you could certainly just do
```cpp
width = widthBits;
height = heightBits;
```
which shows that using separate variables "widthBits" and "heightBits" may not be necessary

Reply:
I will try doing that, will report

Reply:
All done